### PR TITLE
feat(cases): editorial weight + a `featured` search sort for the homepage

### PR DESCRIPTION
**Pull request by notashwinii**
Backend half of the "Case weight/priority field + homepage Featured Cases" card. Adds an editorial `weight` to `Case` and a `featured` sort so the homepage can show curated cases instead of the newest six. The frontend is [Jawafdehi#296](https://github.com/Jawafdehi/Jawafdehi/pull/296) and **must land after this one** — `sort` is a `ChoiceField`, so a frontend that ships first gets a 400 (verified: prod returns `{"sort":["\"featured\" is not a valid choice."]}` today).

## The card was wrong about where this attaches — in two places

**The endpoint.** The card specified `?ordering=-weight` on `CaseViewSet`. That would be dead code: the homepage doesn't call `/api/cases/`. It calls `/api/search/`, and the comment above the query in `Index.tsx` says the switch was deliberate. So the sort has to live in the OpenSearch layer.

**The authoring surface.** The card said staff would "type a number into the Django admin field". Cases are **read-only in Django admin** — all three mutation permissions return `False`, and the comment at `cases/admin.py` is explicit that the SPA `/admin` panel is the sole case *write* surface. Thanks to @coderabbitai for catching this; the first pass shipped a `list_editable` that could never render.

The real write path is `PATCH /api/cases/{id}/` (RFC 6902), which needs four coordinated entries:

| Location | Why it's required |
|---|---|
| `CaseWriteFieldsSerializer` | declares the field. 32-bit bounds because it's an `IntegerField`, not `bigo`'s BigInteger |
| `_build_snapshot` | JSON Patch `replace` fails on a missing path |
| `_PATCH_SCALAR_FIELDS` | **without this the value validates and is silently discarded** — the BB-28 bug the comment in that set warns about |
| `_CREATE_MODEL_FIELDS` | same silent-drop trap on the create path |

Re-indexing needed no work: `_after_commit` already re-indexes explicitly and ungated, because `QuerySet.update()` bypasses `post_save`.

## The sort spec, and why the pre-reindex window is safe

```python
{"weight": {"order": "desc", "missing": 0, "unmapped_type": "integer"}},
{"date":   {"order": "desc", "missing": "_last"}},
{"iri":    {"order": "asc"}},
```

I ran this clause against the **live index** (75 docs, no `weight` in its mapping — `create_index` no-ops on an existing index, so nothing has it until the deploy reindex):

| Clause | Result |
|---|---|
| as above | `200`, top-3 **byte-identical to `sort=newest`** |
| same, `unmapped_type` removed | `400 query_shard_exception` — `No mapping found for [weight] in order to sort on` |

So `unmapped_type` is load-bearing rather than defensive: without it the whole query fails on every index that predates the field, and with it the sort degrades to exactly `newest`. That turns the reindex from a deploy gate into a routine backfill.

- **`date` second, not `created_at`.** While every case is still 0 this fallback *is* the entire order, and `date` is what `newest` sorts by — so the homepage is unchanged until someone assigns a weight (the table above is that claim, measured). The card specified `-created_at`, which would have silently reshuffled every case with a `case_start_date`, breaking its own "untouched cases behave exactly as today" guarantee.
- **`missing: 0`** — a doc indexed before the field existed would otherwise sort *below* an explicitly-weighted-0 case.
- **`iri`** — existing invariant: a non-unique primary can skip/repeat rows under `search_after` paging.

## Two more gaps, found by tracing the field end-to-end

Both are in the second commit; neither was in the original card.

**`weight` never reached a client.** `_serialize_hit` builds a fixed envelope from two hardcoded allowlists, so a `?sort=featured` response could not explain its own order and nothing but `GET /api/cases/{id}/` could read the value back. It now sits in `extra` beside `date`/`date_bs`, whose existing None guard keeps the key absent on pre-backfill docs. Deliberately *not* added to `raw.card` — that payload exists so the SPA renders a card without a second fetch, and no card renders a number.

**The MCP `search_control_plane` sort enum was a stale literal**, so the tool rejected `featured` even though the endpoint behind it accepts it. It stays static (`control_plane_schemas.py` explains why: the schema must build without bootstrapping Django) and is instead pinned to `ALL_SORTS` by a contract test, matching how the other tool schemas track their DRF serializers.

## Two decisions worth reviewer attention

**`weight` is readable in `CaseSerializer`, not just writable.** This reverses the card. The SPA editor renders its form from that payload, so without it an editor sets weights blind — unable to see the current value or what the top case is already at. Side effect: editorial weights are publicly readable, now on `/api/search/` too. `bigo` and `missing_details` already are, so it isn't a new category of exposure; if it's unwanted, the fix is a casework-gated read like `notes` uses (BB-04), which is more work than this PR contains.

**`weight` is excluded from `CaseAdminForm` rather than made `blank=True`.** Under `fields = "__all__"`, a NOT NULL `IntegerField` with a default and no `blank=True` becomes a **required** form field, silently invalidating any `CaseAdminForm` built without it — this broke two admin state-transition e2e tests that have nothing to do with weight. `blank=True` would loosen the model's validation contract everywhere to satisfy a form that cannot write anyway; `readonly_fields` still renders the value from the instance.

## Deploy

1. Merge, then `migrate`.
2. `python manage.py reindex_cases` — **no `--rebuild`**, which drops the index first and would blank the homepage, `/cases` and `/search` while it repopulates. The measurement above is what makes the window before this safe.
3. Confirm `?sort=featured` returns the same six cases as `?sort=newest`, set one weight, confirm it moves to position one and that `extra.weight` reflects it.
4. Then deploy #296.

Note the running mapping will be a dynamically-inferred `long` rather than the declared `integer` (identical for sorting) until some future `--rebuild`. Not worth a destructive rebuild on 75 docs, and not worth a `put_mapping` reconciler that would be the repo's only one.

## Testing

`ruff check` clean (`ruff format` is intentionally not gated here). Full suite: **5074 passed, 5 skipped**. `makemigrations --check` clean.

New coverage: featured sort spec shape, `featured in ALL_SORTS`, featured-past-the-primary equals newest key-for-key, `extra.weight` present/absent on a hit, MCP sort enum vs `ALL_SORTS`, weight doc round-trip (incl. absent and `None`), mapping type, and three PATCH write-path tests (persist + read-back, negative/zero, null and out-of-range rejection).

An earlier revision of this description claimed a pre-existing failure in `tests/mcp/test_http_integration.py::TestProtectedResourceMetadata::test_host_aware_resource_and_scopes`. It passes on this commit; the claim was environmental and is withdrawn.

## Deliberately not included

- **`?ordering=` on `CaseViewSet`** — no consumer now that the homepage is accounted for, and it would need a decision on how `-weight` composes with the existing `_accused_first` tier (which is an `Exists` boolean, not the float an earlier draft of this description called it).
- **CodeRabbit's "require HTTPS for `FRONTEND_BASE_URL`"** (`cases/api_views.py`, flagged outside the diff) — pre-existing from #447, unrelated to weight, and rejecting `http://` would break any local or staging base URL. The setting already defaults to `https://jawafdehi.org`.

## Known limitations

- Unbounded ints over six slots means an editor must know the current top value to beat it. The sortable admin `weight` column and the new `extra.weight` are the mitigation, but "type 9999" drift is a real long-run risk.
- Weight edits are unaudited — `versionInfo` records only state transitions.
- The re-index after a weight PATCH is best-effort and swallowed, so a failed index write still returns 200 with a stale index. Pre-existing behaviour for every scalar edit, not new here.
- With everything at 0 the section renders identically to today, so seeding a few weights alongside the frontend deploy is worthwhile.

## Summary by CodeRabbit

- **New Features**
  - Added editorial weighting to prioritize Featured Cases on the homepage.
  - Added a “Featured” sorting option, ranking higher-weighted cases first, followed by newest date and consistent tie-breakers.
  - Case weights are visible in case responses and can be set or updated through supported case-management interfaces.
- **Validation**
  - Weight values support positive, zero, and negative integers within the supported range.
  - Invalid, null, or out-of-range values are rejected.

### CODE REVIEW

File: cases/admin.py [hunk 1]
```diff
@@ -440,10 +441,18 @@ class Media:
...
         "updated_at",
     ]
 
+    # Editable inline on the changelist: the real workflow is assigning weights to
+    # a handful of cases at once, which would otherwise be one page load + save
+    # each. Legal because ``list_display_links`` is title_with_view_link — Django
+    # rejects an editable field that is also the link column. Sorting by the
+    # ``weight`` column header is also the "what is currently featured?" view.
+    list_editable = ["weight"]
+
```
[the remark this answers is not in the document]
Comment (notashwinii) [reply cases/admin.py]: Good catch — this was correct and it invalidated a premise I'd carried in from the ticket, not just the two lines you flagged.

Confirmed `CaseAdmin.has_change_permission()` returns `False` unconditionally, and the comment above it is explicit that this is by design: cases are read-only in Django admin, and the SPA `/admin` panel is the sole write surface. So `list_editable` could never render, and the `save_model` hook I'd paired with it could never fire. Both removed, along with the now-unused `messages` and `search_index` imports.

Of the three paths you offered I took the third — removed `list_editable` and wired `weight` into the real write path — since granting admin change permission would undo a deliberate architectural decision for one field.

That write path is `PATCH /api/cases/{id}/` and needed four coordinated entries, not one:

- `CaseWriteFieldsSerializer` — declares the field (32-bit bounds, since it's an `IntegerField` rather than `bigo`'s BigInteger)
- `_build_snapshot` — JSON Patch `replace` fails on a missing path
- `_PATCH_SCALAR_FIELDS` — **without this it validates and is silently discarded**, which is the BB-28 bug the comment in that set already warns about
- `_CREATE_MODEL_FIELDS` — the same silent-drop trap on the create path

Re-indexing needed nothing: `_after_commit` already re-indexes explicitly because `QuerySet.update()` bypasses `post_save`.

Two knock-on changes your comment led to, both worth a look:

1. `weight` is now readable in `CaseSerializer`, reversing what the ticket asked for. The SPA editor renders its form from that payload, so without it an editor sets weights blind. This does make editorial weights publicly readable — `bigo` and `missing_details` already are, so not a new category, but flagging it explicitly.

2. `weight` is excluded from `CaseAdminForm` rather than made `blank=True`. Under `fields = "__all__"` a NOT NULL `IntegerField` with a default and no `blank=True` becomes a *required* form field, which silently invalidated any `CaseAdminForm` built without it — that broke two admin state-transition e2e tests unrelated to weight. `blank=True` would have loosened the model's validation contract everywhere to satisfy a form that cannot write; `readonly_fields` still renders the value from the instance.

`weight` stays in `list_display`, where it's display-only and the column sorts — which is the "what's currently featured?" view.

Also added three PATCH write-path tests (persist + read-back, negative/zero, null and out-of-range rejection) so the silent-drop failure mode is guarded rather than just avoided.
